### DISCUSSION TITLE
explanation

### DISCUSSION THREAD

**Discussion by MayoioI** [community/community#134155] [Pull Requests] [answered]
### Select Topic Area

Show & Tell

### Body

Can somebody step by step explain me how my function work?

function changeColor(){
var hex_numbers = ["0","1","2","3","4","5","6","7","8","9","A","B","C","D","F"]
var hexcode = '';

for (var i=0; i < 6; i++){
var random_index = Math.floor(Math.random() * hex_numbers.length); 
hexcode += hex_numbers[random_index]
}

**Comment by Buddhi19** [comment, marked as answer]
Math.random() generates a random decimal number between 0 (inclusive) and 1 (exclusive).
Multiplying by hex_numbers.length (which is 15) scales this random number to the range of the array indices.
Math.floor() rounds down to the nearest integer, giving a valid index for the array.

This loop will iterate 6 times, appending a random number from hex_numbers randomly each time. At the end of the loop hexcode will be a string with 6 numbers corresponding to the 6 characters in a hexadecimal color code (e.g. #RRGGBB).

**Reply by MayoioI** [reply]
ty
